File: db4oj/core/src/com/db4o/internal/BlockSizeBlockConverter.java
/* Copyright (C) 2004 - 2010  Versant Inc.  http://www.db4o.com */

package com.db4o.internal;

import com.db4o.internal.slots.*;

/**
 * @exclude
 */
public final class BlockSizeBlockConverter implements BlockConverter {
	
	private final int _blockSize;

	public BlockSizeBlockConverter(int blockSize) {
		_blockSize = blockSize;
	}
	
    public int bytesToBlocks(long bytes) {
    	return (int) ((bytes + _blockSize -1 )/ _blockSize);
    }
    
    public int blockAlignedBytes(int bytes) {
    	return bytesToBlocks(bytes) * _blockSize;
    }
    
    public int blocksToBytes(int blocks){
    	return blocks * _blockSize;
    }
    
    public Slot toBlockedLength(Slot slot){
    	return new Slot(slot.address(), bytesToBlocks(slot.length()));
    }
    
    public Slot toNonBlockedLength(Slot slot){
    	return new Slot(slot.address(), blocksToBytes(slot.length()));
    }
    

}

File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/BlockSizeBlockConverter.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Internal;
using Db4objects.Db4o.Internal.Slots;

namespace Db4objects.Db4o.Internal
{
	/// <exclude></exclude>
	public sealed class BlockSizeBlockConverter : IBlockConverter
	{
		private readonly int _blockSize;

		public BlockSizeBlockConverter(int blockSize)
		{
			_blockSize = blockSize;
		}

		public int BytesToBlocks(long bytes)
		{
			return (int)((bytes + _blockSize - 1) / _blockSize);
		}

		public int BlockAlignedBytes(int bytes)
		{
			return BytesToBlocks(bytes) * _blockSize;
		}

		public int BlocksToBytes(int blocks)
		{
			return blocks * _blockSize;
		}

		public Slot ToBlockedLength(Slot slot)
		{
			return new Slot(slot.Address(), BytesToBlocks(slot.Length()));
		}

		public Slot ToNonBlockedLength(Slot slot)
		{
			return new Slot(slot.Address(), BlocksToBytes(slot.Length()));
		}
	}
}
